lib/transaction.py

```diff
@@ -653,8 +675,13 @@ def serialize(self, estimate_size=False):
         nLocktime = int_to_hex(self.locktime, 4)
         inputs = self.inputs()
         outputs = self.outputs()
+        lenout=len(outputs)
+        if ((self.op_return is not None) and (self.op_return !="")):
+            lenout+=1
         txins = var_int(len(inputs)) + ''.join(self.serialize_input(txin, self.input_script(txin, estimate_size), estimate_size) for txin in inputs)
-        txouts = var_int(len(outputs)) + ''.join(self.serialize_output(o) for o in outputs)
+        txouts = var_int(lenout) + ''.join(self.serialize_output(o) for o in outputs)
+        if ((self.op_return is not None) and (self.op_return !="")):
```

Comment:
This is just very atypical python usage.

if self.op_return:  is enough.

Also you need to make sure it's always defined in all codepaths.

__init__ () makes no mention of this attribute.  Please set it to none there: self.op_return = None